src/dra/7E4BC.c

```diff
@@ -892,7 +892,213 @@ void EntityHitByIce(Entity* self) {
 
 // Transparent white circle closes over player
 // Entity ID #38. Created by blueprint 53. No known callers.
-INCLUDE_ASM("dra/nonmatchings/7E4BC", EntityTransparentWhiteCircle);
+// Near-duplicate of RIC EntityShrinkingPowerUpRing.
+void EntityTransparentWhiteCircle(Entity* self) {
+    s16 selfX;
+    s16 selfY;
+    s16 rScale;
+    s16 gScale;
+    s16 bScale;
+    s16 gOffset;
+    s16 bOffset;
+    s16 sp38;
+    Primitive* prim1;
+    Primitive* prim2;
+    s32 baseAngle;
+    s32 i;
+    s16* loadedParams;
+    s32 temp_s6;
+
+    u32 upperparams = self->params & 0x7F00;
+
+    loadedParams = &D_800ADDE8[(upperparams >> 8) & 0x3F][0];
+    rScale = loadedParams[2];
+    gScale = loadedParams[3];
+    bScale = loadedParams[4];
+    gOffset = loadedParams[0];
+    bOffset = loadedParams[1];
+    self->posX.i.hi = PLAYER.posX.i.hi;
+    self->posY.i.hi = PLAYER.posY.i.hi;
+    temp_s6 = (upperparams >> 8) & 0x40;
+    func_8010DFF0(1, 1);
+    switch (self->step) {
+    case 0:
+        self->primIndex = AllocPrimitives(PRIM_GT4, 32);
+        if (self->primIndex == -1) {
+            DestroyEntity(self);
+            return;
+        }
+        self->flags = FLAG_UNK_04000000 | FLAG_HAS_PRIMS | FLAG_UNK_40000 |
+                      FLAG_UNK_20000 | FLAG_UNK_10000;
+        prim2 = prim1 = &g_PrimBuf[self->primIndex];
+        for (i = 0; i < 16; i++) {
+            prim1 = prim1->next;
+        }
+        for (i = 0; i < 16; i++) {
+            prim2->u0 = ((rcos((s16)(i << 8)) * 2) >> 8) + 0x20;
+            prim2->v0 = -((rsin((s16)(i << 8)) * 2) >> 8) - 0x21;
+            prim2->u1 = ((rcos((s16)(i + 1 << 8)) * 2) >> 8) + 0x20;
+            prim2->v1 = -((rsin((s16)(i + 1 << 8)) * 2) >> 8) - 0x21;
+            prim1->u2 = prim1->u3 = 0x20;
+            prim1->v2 = prim1->v3 = 0xDF;
+            prim2->u2 = prim1->u0 = (prim2->u0 + prim1->u2) / 2;
+            prim2->v2 = prim1->v0 = (prim2->v0 + prim1->v2) / 2;
+            prim2->u3 = prim1->u1 = (prim2->u1 + prim1->u3) / 2;
+            prim2->v3 = prim1->v1 = (prim2->v1 + prim1->v3) / 2;
+            prim1->tpage = prim2->tpage = 0x1A;
+            prim1->clut = prim2->clut = 0x15F;
+            prim1->priority = prim2->priority = PLAYER.zPriority + 2;
+            prim1->drawMode = prim2->drawMode = 0x235;
```

Comment:
drawmode enum

Reply:
Whoops! you'd think I'd remember after https://github.com/Xeeynamo/sotn-decomp/pull/1379 😅